client/my-sites/subscribers/components/subscribers-page/subscribers-page-context.tsx

```diff
@@ -37,16 +43,21 @@ const DEFAULT_PER_PAGE = 10;
 export const SubscribersPageProvider = ( {
 	children,
 	siteId,
-	page,
+	filterOption = SubscribersFilterBy.All,
+	pageNumber,
+	searchTerm,
+	sortTerm = SubscribersSortBy.DateSubscribed,
+	filterOptionChanged,
 	pageChanged,
+	searchTermChanged,
+	sortTermChanged,
 }: SubscribersPageProviderProps ) => {
-	const [ searchTerm, setSearchTerm ] = useState( '' );
 	const [ perPage, setPerPage ] = useState( DEFAULT_PER_PAGE );
-	const [ sortTerm, setSortTerm ] = useState( SubscribersSortBy.DateSubscribed );
-	const [ filterOption, setFilterOption ] = useState( SubscribersFilterBy.All );
 
 	const handleSearch = useCallback( ( term: string ) => {
-		setSearchTerm( term );
+		searchTermChanged( term );
+		pageChanged( 1 );
```

Comment:
Instead of calling `pageChanged( 1 );` we need to call `pageClickCallback( 1 );`:

```suggestion
		pageClickCallback( 1 );
```

The reason for this is that when `pageChanged( 1 );` is used, the `useEffect` inside [use-pagination.ts](https://github.com/Automattic/wp-calypso/blob/9681ad8ca0d2b9af9222f02968ebe4e7c378202b/client/my-sites/subscribers/hooks/use-pagination.ts) will get called:

```jsx
	useEffect( () => {
		if ( ! isFetching && currentPage !== page ) {
			pageChanged( currentPage );
		}
	}, [ currentPage, isFetching, page, pageChanged ] );
```

and since `currentPage !== page`, after a brief moment when the page is set to `1`, it gets set back to the original page number instead.

In other words, `pageChanged( 1 );` does not have any effect over the `currentPage` state set inside [use-pagination.ts](https://github.com/Automattic/wp-calypso/blob/9681ad8ca0d2b9af9222f02968ebe4e7c378202b/client/my-sites/subscribers/hooks/use-pagination.ts).

With that said, we could also consider renaming `pageClickCallback` to `pageChangeCallback` for instance (as it will handle not just clicking on the numbers in the pager, but also search).

Reply:
Since Paulo will be AFK, I have addressed these changes in https://github.com/Automattic/wp-calypso/pull/79039/commits/e7862a386c485f1418190cf292eb6ff24ad08c6e.